source/clear-linux/tutorials/kubernetes.rst

```diff
@@ -0,0 +1,303 @@
+
+Kubernetes and a set of supported :abbr:`CRI (Container Runtime Interface)` 
+runtimes are included in the `cloud-native-basic`_ bundle. To install the 
+framework, enter the following command:
+
+.. code-block:: bash
+
+   sudo swupd bundle-add cloud-native-basic
+
+Configure Kubernetes
+********************
+
+This tutorial uses the basic default Kubernetes configuration for simplicity.
+You must define your Kubernetes configuration according to your specific
+deployment and your security needs.
+
+#. Enable IP forwarding to avoid "preflight check" errors:
+
+   Create (or edit if it exists) the file :file:`/etc/sysctl.d/60-k8s.conf`
+   and include the following line:
+
+   .. code-block:: bash
+
+      net.ipv4.ip_forward = 1
+
+   Apply the change:
+
+   .. code-block:: bash
+
+      sudo systemctl restart systemd-sysctl
+
+#. Enable kubelet service:
+
+   .. code-block:: bash
+
+      sudo systemctl enable kubelet.service
+
+#. Disable swap using one of the following methods, either:
+
+   a) Temporarily:
+
+      .. code-block:: bash
+
+         sudo swapoff -a
+
+      .. note::
+
+         Swap will be enabled at next reboot, causing failures in
+         your cluster.
+
+   or:
+
+   b) Permanently:
+
+      Mask the swap partition:
+
+      .. code-block:: bash
+
+         sudo systemctl mask `sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps
+         sudo swapoff -a
+
+      .. note::
+
+         On systems with limited resources, some performance degradation may
+         be observed while swap is disabled.
+
+#. Configure the Kubernetes runtime interface, either:
+
+   a) Run Kubernetes with Docker + runc:
+
+      #. Enable docker.service:
+
+         .. code-block:: bash
+
+            sudo systemctl enable docker.service
+
+      #. Create (or edit if it exists) the file :file:`/etc/systemd/system/docker.service.d/51-runtime.conf` and include the following lines:
+
+         .. code-block:: bash
+
+            [Service]
+            Environment="DOCKER_DEFAULT_RUNTIME=--default-runtime runc"
+
+      #. Create (or edit if it exists) the file :file:`/etc/systemd/system/kubelet.service.d/10-kubeadm.conf` and include the following lines:
+
+         .. code-block:: bash
+
+            [Service]
+            Environment="KUBELET_EXTRA_ARGS="
+
+      #. Restart services:
+
+         .. code-block:: bash
+
+            sudo systemctl daemon-reload
+            sudo systemctl restart docker
+            sudo systemctl restart kubelet
+
+   or:
+
+   b) Run Kubernetes with CRI-O + your desired runtime. You can use multiple
+      runtimes with CRI-O, including *runc* and *kata-runtime*. To use
+      CRI-O + *kata-runtime*:
+
+      #. Enable crio.service:
+
+         .. code-block:: bash
+
+            sudo systemctl enable crio.service
+
+      #. Restart services:
+
+         .. code-block:: bash
+
+            sudo systemctl restart crio
+            sudo systemctl restart kubelet
+
```

Comment:
@eadamsintel  Do you think adding a directory path to access and verify the `hostname` is needed here? 
If yes, we can provide: `/usr/share/defaults/etc/hosts`

I'd like to know what the value-add is. Maybe I missed that part of the conversation previously.

Reply:
@mvincerx kubelet code explicitly refers to file ```/etc/hosts```